library/std/src/backtrace.rs

```diff
@@ -251,10 +252,17 @@ impl Backtrace {
                 Err(_) => false,
             },
         };
-        ENABLED.store(enabled as usize + 1, SeqCst);
+        Self::set_enabled(enabled);
         enabled
     }
 
+    /// Explicitly enable or disable backtrace capturing, overriding the default
+    /// derived from the `RUST_BACKTRACE` or `RUST_LIB_BACKTRACE` environment
+    /// variables.
+    pub fn set_enabled(enabled: bool) {
```

Comment:
This API doesn't look sufficient enough to handle the full functionality of what `RUST_BACKTRACE` can do, as `RUST_BACKTRACE` has multiple values, and might even be extended to more in the future. I think it would be good to at least change this to take an enum, something like this:
```rust
#[non_exhaustive]
pub Enum {
    Off,
    Short,
    Full,
}
```

Also, I think that this API should be "set at most once," similar to `log::set_logger`, as an application like the one I maintain doesn't want a (third-party) library to change this value after we set it in `main()`. (The ability to lock the backtrace level is another motivation for adding this new API.)